### PR TITLE
Date_last_login fix and add link to notify base

### PR DESCRIPTION
**Pull request by hmoco**

### CODE REVIEW

File: website/templates/emails/notify_base.mako [hunk 1]
```diff
@@ -43,7 +43,7 @@
                 <tbody>
                     <tr>
                         <td style="border-collapse: collapse;">
-                            <p class="small text-center" style="text-align: center;font-size: 12px;">To change your email notification preferences, visit your project or user settings.</p>
+                            <p class="small text-center" style="text-align: center;font-size: 12px;">To change your email notification preferences, visit your project or <a href="http://osf.io/settings">user settings</a>.</p>
```
Comment (hmoco) [inline website/templates/emails/notify_base.mako]: @sloria Please take a look at this, not sure if there is a better way to get what is wanted in https://openscience.atlassian.net/browse/OSF-4142?focusedCommentId=23888&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23888 while still using the base template.

Comment (sloria) [inline website/templates/emails/notify_base.mako]: Shouldn't this link to `/settings/notifications/`? Also, can you avoid hardcoding the domain, so that this will work on staging, localhost, etc.?

File: framework/sessions/__init__.py [hunk 2]
```diff
@@ -171,5 +171,10 @@ def before_request():
 def after_request(response):
     if session.data.get('auth_user_id'):
         session.save()
-
+        from framework.auth.core import _get_current_user
```
Comment (sloria) [inline framework/sessions/__init__.py]: Is this necessary to avoid a circular import? If so, add a comment explaining that.

**Comment by sloria** [comment]
Pass finished. Can you also look into the failing tests?

File: website/templates/emails/notify_base.mako (hunk 1 above)
Comment (hmoco) [inline website/templates/emails/notify_base.mako]: Yes and I don't think so, since its a base template I cannot be sure where the mako variables will be populated, so I could for my engagement mails, but it would throw a mako error for any other template using it.

**Comment by hmoco** [comment]
@sloria Responded to comments

**Comment by sloria** [comment]
@hmoco Can you update this with the latest develop? That should fix the failing build.

File: framework/sessions/__init__.py [hunk 3]
```diff
@@ -171,5 +172,14 @@ def before_request():
...
         session.save()
+        #import here to prevent circular import error
+        from framework.auth.core import _get_current_user
 
+        user = _get_current_user()
+        if user:
+            try:
+                user.date_last_login = datetime.utcnow()
+                user.save()
+            except:
```
Comment (sloria) [inline framework/sessions/__init__.py]: Why is this necessary? What exception might be raised here?

Comment (hmoco) [inline framework/sessions/__init__.py]: If theres an error in the request, it gets called again in save I believe, which causes tests that use requests and expect a failure to error instead of sending the failure as part of the response.

Comment (sloria) [inline framework/sessions/__init__.py]: Hm, then maybe it would make sense `date_last_login`  in a `before_request` hook, before any work is done in the view?

**Comment by hmoco** [comment]
@sloria I think this might have done the trick

File: framework/sessions/__init__.py [hunk 4]
```diff
@@ -150,6 +151,8 @@ def before_request():
             session.data['auth_user_username'] = user.username
             session.data['auth_user_id'] = user._primary_key
             session.data['auth_user_fullname'] = user.fullname
+            user.date_last_login = datetime.utcnow()
+            user.save()
```
Comment (hmoco) [inline framework/sessions/__init__.py]: @sloria, going to do a functional test on this, pushing so Travis can do his thing.

Comment (sloria) [inline framework/sessions/__init__.py]: :+1:

**Comment by hmoco** [comment]
Should be good to go @sloria

File: framework/auth/core.py [hunk 5]
```diff
@@ -367,7 +367,7 @@ class User(GuidStoredObject, AddonModelMixin):
     # hashed password used to authenticate to Piwik
     piwik_token = fields.StringField()
 
-    # date the user last logged in via the web interface
+    # date the user last sent a request
```
Comment (sloria) [inline framework/auth/core.py]: :+1:

File: framework/sessions/__init__.py (hunk 4 above)
Comment (sloria) [inline framework/sessions/__init__.py]: Would be good to have a test for this field being updated on an authenticated request. This can happen in a separate PR, since this PR is blocking QA.
